examples/admins/main.tf

```diff
@@ -0,0 +1,33 @@
+module "naming" {
+  source  = "cloudnationhq/naming/azure"
+  version = "~> 0.1"
+
+  suffix = ["demo", "dev"]
+}
+
+module "rg" {
+  source  = "cloudnationhq/rg/azure"
+  version = "~> 0.1"
+
+  groups = {
+    demo = {
+      name   = module.naming.resource_group.name
+      region = "westeurope"
+    }
+  }
+}
+
+data "azurerm_client_config" "current" {}
+
+module "kv" {
+  source  = "cloudnationhq/kv/azure"
+  version = "~> 0.1"
+  vault = {
+    name          = module.naming.key_vault.name_unique
+    location      = module.rg.groups.demo.location
+    resourcegroup = module.rg.groups.demo.name
+
+    admins                    = [data.azurerm_client_config.current.object_id]
```

Comment:
Using a list builds up a index, so if you have a list of admins and remove one, it will replace all the others as well. It is better to use a map in this case.

Reply:
down in the expression tree a foreach is used and the key is the same as the value. I believe this is good enough.